compiler/modules/front.syntax/parseTypeDesc.bal

```diff
@@ -166,9 +156,18 @@ function parsePrimaryTypeDesc(Tokenizer tok) returns TypeDesc|err:Syntax {
             // JBUG should not need cast #30191
             return <LeafTypeDesc>cur;
         }
+        var prefix if prefix == "string" || prefix == "int" => {
```

Comment:
Can't I just do `"string"|"int"` here?

Reply:
Type narrowing doesn't quite work, we can do:
```ballerina
        "string"
        |"int" => {
            "int"|"string" prefix = <"int"|"string"> cur;
            Position pos = tok.currentPos();
            check tok.advance();
            if tok.current() != ":" {
                return prefix;
            }
            check tok.advance();
            return { prefix, typeName: check tok.expectIdentifier(), pos };
        }
```

Reply:
```suggestion
        var prefix if prefix is "string"|"int" => {
```

Reply:
Went with `LeafTypeDesc`